src/components/directory-content.js

```diff
@@ -120,6 +120,8 @@ const DirectoryContent = () => {
     const [selectedUuids, setSelectedUuids] = useState(new Set());
 
     const currentChildren = useSelector((state) => state.currentChildren);
+    const [displayedElements, setDisplayedElements] = useState([]);
```

Comment:
I'm not sure you need another state here (displayedElements). You already have currentChildren and it creates confusion.
You may should complete `currentChildren` with matching (same directory) `uploadingStudies` in the `directory-container` component which synchronize data when browsing and everything will work as before with `currentChildren` in here.

Reply:
current children are distant elements, and are selectable uploading ones are not, also when currentChildren changes we fetch associated metadata (useEffect),

Reply:
made the change